nasa-gibs/worldview
# WV-2084 Allow override of About config

## Description
A user asked to make the About be overridden so they can change the text when spinning up their own copy.  The user asked to have each section in the About page to be overridden separately.

## Fixes
The only way to edit the About page was to edit the `about-page.js` itself.  In an effort to leave the default descriptions, separate JS files for each About section was created.  These individual JS pages hold the custom text and are named for their respective sections.

Once the custom text is filled out, they need to be activated by changing the respective toggles from `false` to `true`.

```javascript
// "Welcome" section will be overridden with the 
// description in the "welcome-custom.js" file
const overrideWelcome = true; 

// "Keyboard Shortcuts" section will use the 
// default description set in the "about-page.js"
const overrideKeyboardShortcuts = false; 
```
## How To Test
After pulling down the branch, run the following commands to get started:
```
npm ci
npm run build
npm run watch:debug:devtools  // This allows hot reloading on file save
```
1. Open `about-page.js` and any of the `[section]-custom.js` files. Example: `welcome-custom.js`
2. Make some changes to the text in `welcome-custom.js`. 
3. In `about-page.js`, change the `overrideWelcome` variable to `true`.
4. Open [http://localhost:3000](http://localhost:3000) in your browser.
5. Close the welcome modal, click on the Information button ("i" button) at the top right of the page and click "About".
6. You should _only_ see the description/text that you entered in `welcome-custom.js` file.  The original text defined in `about-page.js` should not be there.
